### DISCUSSION TITLE
Only once render  to blade

### DISCUSSION THREAD

**Discussion by sueboy19** [livewire/livewire#7655] [3. Feature ideas]
Little like this  
https://github.com/livewire/livewire/discussions/7242  

Now Accounting project have 2,xxx subject records. When I create voucher, need pickup subject and input money.
TomeSelect search 2,xxx subjects.
```php
public $subjects; // At mount() put 2,xxx subject records
```

## Problem
Every time wire be triggered, /update response subjects data to front. Transmit subjects data size near 2M.
![image](https://github.com/livewire/livewire/assets/6514381/ec459a9d-b7c4-4430-aaee-9b7e7659659a)

## Now
Alpine x-init 
```js
  subjects: {!! json_encode($subjects) !!},
```
But every time wire be triggered that still transmit subjects datas.

## Other Way
Write API for subjects. Alpine x-init call this subjects api. Transmit to front only once.

## Feature
Wish send subjects to front only once. Any wire be triggered, don't reponse again. 
Pass Hydration or without Hydration.

## Performance
The Best. 100x up.

Thanks.

**Comment by PhiloNL** [comment]
Do you need all of those records on a public property / do you need the properties to be interactive?
I recommend using computed properties whenever possible and taking a different approach when dealing with a large number of records.

**Reply by sueboy19** [reply]
> Do you need all of those records on a public property / do you need the properties to be interactive?

Accounting work need to create Voucher that choose subjects and input money. Accountant typeing some keyword, tom select search and dropdown list for pickup. One voucher maybe 20~30 items need to add or modify. 

So need send all subjects to front, subjects are fixed. No need interactive. 

I know better way is like Computed. If need to refresh can use unset(). Have both fixed and refresh. This Accounting project other function already use Computed. But size and reponse time is littel problem.

I try to use #[Computed],  transmit subjects data size still 2MB.

![image](https://github.com/livewire/livewire/assets/6514381/9e0f2430-b0fe-471b-a4ff-2ecdeb3ee905)


```php
    #[Computed]
    public function subjects()
    {
        return $this->Subjects();
    }
```

blade
```blade
@foreach($this->subjects as $subject)
{
 ....
```

So #[Computed] can't avoid it.

Project some function have more than 20MB situation. 
Two way:  A. use multiple page to limit size. One page works split to many pages. B. try to clean public variable when no need.

Thanks your help.

**Comment by sueboy19** [comment]
# New Way

Use wire:Stream with Alpine.js x-init

```php
    public $answer;
    public function ded()
    {
        $this->stream(to: 'answer',
            content: json_encode(DB::table('answers')->get())
        );
    }
```

```blade
<div class="container" x-data x-init="await $wire.ded(); console.log(JSON.parse($refs.cc.textContent));">
    <div x-ref="cc" wire:stream="answer" wire:ignore x-cloak>
        {{ $answer }}
    </div>

     ...
</div>
```

Now subjects data are not send again. Only send once to front.
